libjolie/src/jolie/lang/parse/OLParser.java

```diff
@@ -811,8 +812,160 @@ private void parseInterfaceOrPort()
 				node = null;
 			}
 		}
-	}
+    }
+
+    private OutputPortInfo createInternalServicePort(String name, List< InterfaceDefinition> interfaceList)
+        throws ParserException
+    {
+        OutputPortInfo p = new OutputPortInfo(getContext(), name);
+        
+        for (InterfaceDefinition interfaceDefinition : interfaceList) {
+            interfaceDefinition.copyTo(p);
+        }
+        return p;
+    }
+    
+    private InputPortInfo createInternalServiceInputPort(String serviceName, List< InterfaceDefinition > interfaceList)
+        throws ParserException
+    {
+        OLSyntaxNode protocolConfiguration = new NullProcessStatement( getContext() );
+        InputPortInfo iport = null;
+        try {
+            iport = new InputPortInfo(
+                getContext(),
+                serviceName + "InputPort", //input port name
+                new URI( Constants.LOCAL_LOCATION_KEYWORD ),
+                null,
+                protocolConfiguration,
+                new InputPortInfo.AggregationItemInfo[] {},
+                new HashMap<String, String>() );
+            
+            for( InterfaceDefinition i : interfaceList ) {
+                i.copyTo(iport);
+        	}                    
+        } catch (URISyntaxException e) {
+             throwException( e );
+        }
+        return iport;
+    }
+    
+    /**
+     * Parses an internal service, i.e. service service_name {}
+     * 
+     * @throws IOException
+     * @throws ParserException 
+     */
+    private void parseInternalService()
+        throws IOException, ParserException
+    {
+        //only proceed if a service declaration
+        if (!token.isKeyword("service")) return;
+        
+		//get service name
+        getToken();
+        assertToken(Scanner.TokenType.ID, "expected service name");
+        String serviceName = token.content();
+        
+        //validate token
+        getToken();
+        eat(Scanner.TokenType.LCURLY, "{ expected");
+        
+        //initialize internal interface and interface list
+        List< InterfaceDefinition > interfaceList = new ArrayList< InterfaceDefinition >();
+        
+        OLSyntaxNode internal_main = null;
+        SequenceStatement internal_init = null;
```

Comment:
internal_init could probably just be OLSyntaxNode, SequenceStatement seems redundant.